src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -248,20 +249,43 @@ export class Visualization {
     this.forceSim.updateRelationships(this.graph)
   }
 
-  private handleZoomClick(zoomType: ZoomType): void {
+  private handleZoomClick = (zoomType: ZoomType): void => {
     this.draw = true
     this.isZoomClick = true
 
     if (zoomType === ZoomType.IN) {
-      this.zoomBehavior.scaleBy(this.root, 1.3)
+      this.zoomTo(1.3)
```

Comment:
Why not leave these functions as they are and just use the new function for when we have a point to zoom to?

Reply:
Since the function seems to be doing two different things now

Reply:
Sure, I was thinking to have the same abstraction level of the different cases, but I can see that it doesn't really help here since the `this.zoomBehavior.scaleBy(this.root, 1.3)` is pretty easy to read as it is. I will create a zoomToFit() function instead and put all the logic for that there and removing the zoomTo function.